internal/provider/security/security_certificate_data_source.go

```diff
@@ -0,0 +1,172 @@
+package security
+
+import (
+	"context"
+	"fmt"
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/provider/connection"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/interfaces"
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/utils"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces
+var _ datasource.DataSource = &SecurityCertificateDataSource{}
+
+// NewSecurityCertificateDataSource is a helper function to simplify the provider implementation.
+func NewSecurityCertificateDataSource() datasource.DataSource {
+	return &SecurityCertificateDataSource{
+		config: connection.ResourceOrDataSourceConfig{
+			Name: "security_certificate",
+		},
+	}
+}
+
+// SecurityCertificateDataSource defines the data source implementation.
+type SecurityCertificateDataSource struct {
+	config connection.ResourceOrDataSourceConfig
+}
+
+// SecurityCertificateDataSourceModel describes the data source data model.
+type SecurityCertificateDataSourceModel struct {
+	CxProfileName types.String `tfsdk:"cx_profile_name"`
+	Name          types.String `tfsdk:"name"`
+	CommonName    types.String `tfsdk:"common_name"`
+	SVMName       types.String `tfsdk:"svm_name"`
+	Scope         types.String `tfsdk:"scope"`
+	Type          types.String `tfsdk:"type"`
+	SerialNumber  types.String `tfsdk:"serial_number"`
+	CA            types.String `tfsdk:"ca"`
+	HashFunction  types.String `tfsdk:"hash_function"`
+	KeySize       types.Int64  `tfsdk:"key_size"`
+	ExpiryTime    types.String `tfsdk:"expiry_time"`
+	ID            types.String `tfsdk:"id"`
+}
+
+// Metadata returns the data source type name.
+func (d *SecurityCertificateDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + d.config.Name
+}
+
+// Schema defines the schema for the data source.
+func (d *SecurityCertificateDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "SecurityCertificate data source",
+
+		Attributes: map[string]schema.Attribute{
+			"cx_profile_name": schema.StringAttribute{
+				MarkdownDescription: "Connection profile name",
+				Required:            true,
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "The unique name of the security certificate per SVM.",
+				Optional:            true,
+			},
+			"common_name": schema.StringAttribute{
```

Comment:
Is the common_name unique in the cluster?

Reply:
I just checked in my VSIM cluster with 2 nodes where out of 90+ certificates the below one is duplicated:
```
"type": "server",
"common_name": "testtest",
```

Reply:
Then need to think about in what kind of condition will get one record since this is a data source not a list of data source**s**.